clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -3307,6 +3307,29 @@ void Sema::CheckShadowInheritedFields(const SourceLocation &Loc,
   }
 }
 
+template <typename AttrType>
+inline static bool HasAttribute(const QualType &T) {
+  if (const TagDecl *TD = T->getAsTagDecl())
+    return TD->hasAttr<AttrType>();
+  if (const TypedefType *TDT = T->getAs<TypedefType>())
+    return TDT->getDecl()->hasAttr<AttrType>();
+  return false;
+}
+
+inline static bool IsUnusedPrivateField(FieldDecl *FD) {
```

Comment:
The `inline` in the static functions should go I think, and I hope the `FieldDecl` here can be `const`.

Reply:
Thanks for the feedback. I've added changes.